### PR TITLE
added logging

### PR DESCRIPTION
**Pull request by EgorShtab**

### CODE REVIEW

**Review by Sauric — APPROVED** [review]

File: src/api/Api.Test/Api.Test.csproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.0" />
```
Comment (pmosk) [inline src/api/Api.Test/Api.Test.csproj:15]: Remove this dependency

File: src/api/Api/Api.csproj [hunk 2]
```diff
@@ -4,13 +4,14 @@
     <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
     <RootNamespace>GGroupp.Infra</RootNamespace>
     <AssemblyName>GGroupp.Infra.Sql.Api</AssemblyName>
-    <Version>1.3.0</Version>
+    <Version>1.3.1</Version>
```
Comment (pmosk) [inline src/api/Api/Api.csproj:7]: Set v1.4.0

File: src/api/Api/SqlApi/SqlApi.cs [hunk 3]
```diff
@@ -2,38 +2,66 @@
...
 using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Logging;
 
 namespace GGroupp.Infra;
 
 internal sealed partial class SqlApi : ISqlApi
 {
-    public static SqlApi Create(IDbProvider dbProvider)
+    public static SqlApi Create(IDbProvider dbProvider, bool needLogging, ILoggerFactory? loggerFactory)
```
Comment (pmosk) [inline src/api/Api/SqlApi/SqlApi.cs:12]: Not to use needLogging here. If loggerFactory is null logger will be null

File: src/api/Api/SqlApi/SqlApi.cs [hunk 4]
```diff
@@ -2,38 +2,66 @@
...
+    private SqlApi(IDbProvider dbProvider, ILogger? logger)
         =>
-        this.dbProvider = dbProvider;
+        (this.dbProvider, this.logger) = (dbProvider, logger);
 
     private DbCommand CreateDbCommand(DbConnection dbConnection, IDbQuery query)
     {
         var dbCommand = dbConnection.CreateCommand();
         dbCommand.CommandText = query.GetSqlQuery();
+        var parametersLogBuilder = new StringBuilder();
```
Comment (pmosk) [inline src/api/Api/SqlApi/SqlApi.cs:30]: var parameterLogBuilder = logger is null ? null : new StringBuilder();

File: src/api/Api/SqlApi/SqlApi.cs [hunk 5]
```diff
@@ -2,38 +2,66 @@
...
             dbCommand.CommandTimeout = query.TimeoutInSeconds.Value;
         }
 
+        logger?.LogDebug("SQL: {0}, Parameters: {1}", dbCommand.CommandText, parametersLogBuilder.ToString());  
+
         return dbCommand;
+
+        void AppendParameter(StringBuilder builder, DbParameter dbParameter)
+        {
+            if (logger is null)
```
Comment (pmosk) [inline src/api/Api/SqlApi/SqlApi.cs:49]: if (parameterLogBuilder is null)

File: src/api/Api/SqlApi/SqlApi.cs [hunk 6]
```diff
@@ -2,38 +2,66 @@
...
         if (query.TimeoutInSeconds is not null)
         {
             dbCommand.CommandTimeout = query.TimeoutInSeconds.Value;
         }
 
+        logger?.LogDebug("SQL: {0}, Parameters: {1}", dbCommand.CommandText, parametersLogBuilder.ToString());  
+
         return dbCommand;
+
+        void AppendParameter(StringBuilder builder, DbParameter dbParameter)
```
Comment (pmosk) [inline src/api/Api/SqlApi/SqlApi.cs:47]: You can use only one parameter: void AppendParameter(DbParameter dbParameter)

**Review by pmosk — CHANGES_REQUESTED** [review]

**Review by pmosk — APPROVED** [review]

**Pull request merged**